tests/components/tasmota/test_device_trigger.py

```diff
@@ -0,0 +1,832 @@
+"""The tests for MQTT device triggers."""
+import copy
+import json
+
+from hatasmota.switch import TasmotaSwitchTriggerConfig
+import pytest
+
+import homeassistant.components.automation as automation
+from homeassistant.components.tasmota.const import DEFAULT_PREFIX, DOMAIN
+from homeassistant.components.tasmota.device_trigger import async_attach_trigger
+from homeassistant.setup import async_setup_component
+
+from .test_common import DEFAULT_CONFIG
+
+from tests.async_mock import patch
+from tests.common import (
+    assert_lists_same,
+    async_fire_mqtt_message,
+    async_get_device_automations,
+)
+
+
+async def test_get_triggers(hass, device_reg, entity_reg, mqtt_mock, setup_tasmota):
+    """Test we get the expected triggers from a discovered mqtt device."""
+    config = copy.deepcopy(DEFAULT_CONFIG)
+    config["swc"][0] = 0
+    mac = config["mac"]
+
+    async_fire_mqtt_message(hass, f"{DEFAULT_PREFIX}/{mac}/config", json.dumps(config))
+    await hass.async_block_till_done()
+
+    device_entry = device_reg.async_get_device(set(), {("mac", mac)})
+    expected_triggers = [
+        {
+            "platform": "device",
+            "domain": DOMAIN,
+            "device_id": device_entry.id,
+            "discovery_id": "00000049A3BC_switch_1_TOGGLE",
+            "type": "button_short_press",
+            "subtype": "switch_1",
+        },
+    ]
+    triggers = await async_get_device_automations(hass, "trigger", device_entry.id)
+    assert_lists_same(triggers, expected_triggers)
+
+
+async def test_get_unknown_triggers(
+    hass, device_reg, entity_reg, mqtt_mock, setup_tasmota
+):
+    """Test we don't get unknown triggers."""
+    # Discover a device without device triggers
+    config = copy.deepcopy(DEFAULT_CONFIG)
+    config["swc"][0] = -1
+    mac = config["mac"]
+
+    async_fire_mqtt_message(hass, f"{DEFAULT_PREFIX}/{mac}/config", json.dumps(config))
+    await hass.async_block_till_done()
+
+    device_entry = device_reg.async_get_device(set(), {("mac", mac)})
+
+    assert await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)